pkg/operator/ceph/csi/peermap/config.go

```diff
@@ -0,0 +1,337 @@
+	mappings *PeerIDMappings,
+) error {
+	ctx := context.TODO()
+	data, err := mappings.toString()
+	if err != nil {
+		return errors.Wrap(err, "failed to convert peer cluster mappings struct to string")
+	}
+
+	opNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	if opNamespace == "" {
+		return errors.New("failed to get rook-operator namespace")
+	}
+
+	request := types.NamespacedName{Name: mappingConfigName, Namespace: opNamespace}
+	existingConfigMap := &v1.ConfigMap{}
+
+	err = clusterContext.Client.Get(ctx, request, existingConfigMap)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Create new configMap
+			return createConfig(clusterContext, request, data)
+		}
+		return errors.Wrapf(err, "failed to get existing mapping configMap %q", existingConfigMap.Name)
+	}
+
+	// Update existing configMap
+	existingConfigMap.Data[mappingConfigkey] = data
+	if err := clusterContext.Client.Update(ctx, existingConfigMap); err != nil {
+		return errors.Wrapf(err, "failed to update existing mapping configMap %q", existingConfigMap.Name)
+	}
+
+	return nil
+}
+
+func createConfig(clusterContext *clusterd.Context, request types.NamespacedName, data string) error {
+	newConfigMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      request.Name,
+			Namespace: request.Namespace,
+		},
+		Data: map[string]string{
+			mappingConfigkey: data,
+		},
+	}
+
+	// Get Operator owner reference
+	operatorPodName := os.Getenv(k8sutil.PodNameEnvVar)
+	ownerRef, err := k8sutil.GetDeploymentOwnerReference(clusterContext.Clientset, operatorPodName, request.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to get operator owner reference")
+	}
+	if ownerRef != nil {
+		blockOwnerDeletion := false
+		ownerRef.BlockOwnerDeletion = &blockOwnerDeletion
+	}
+
+	ownerInfo := k8sutil.NewOwnerInfoWithOwnerRef(ownerRef, request.Namespace)
+
+	// Set controller reference only when creating the configMap for the first time
+	err = ownerInfo.SetControllerReference(newConfigMap)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference on configMap %q", newConfigMap.Name)
+	}
+
+	err = clusterContext.Client.Create(context.TODO(), newConfigMap)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create mapping configMap %q", newConfigMap.Name)
+	}
+	return nil
+}
+
+func decodePeerToken(token string) (*cephclient.PeerToken, error) {
+	// decode the base64 encoded token
+	decodedToken, err := base64.StdEncoding.DecodeString(string(token))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode bootstrap peer token")
+	}
+
+	// Unmarshal the decoded token to a Go type
+	var decodedTokenToGo cephclient.PeerToken
+	err = json.Unmarshal(decodedToken, &decodedTokenToGo)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal decoded token")
+	}
+
+	logger.Debugf("peer cluster info %+v", decodedTokenToGo)
+
+	return &decodedTokenToGo, nil
+}
+
+func createTmpFile(content, pattern string) (*os.File, error) {
```

Comment:
Please use or refact. `CreateTempPasswordFile` from the mgr package.

Reply:
Moved `CreateTempPasswordFile` to `pkg/util/file` and using it as `CreateTempFile`